Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -554,23 +559,78 @@ int xyz(char *dst) {
 }
 ```
 
- 2. _The file storage related integrity checks:_ 
+##### Sample Implementation - Storage
 
-##### Sample Implementation - application-source
+When providing integrity on the application storage itself, you can either create an HMAC or a signature over a given key-value pair or over a file stored on the device. When you create an HMAC, it is best to use hte CommonCrypto implementation.
+In case of the need for encryption: Please make sure that you encrypt and then HMAC as described in [1].
 
-##### Sample Implementation - Storage
+When generating an HMAC with CC:
+
+1. get the data as `NSData`.
+2. Get the data key (possibly from the keychain)
+3. Calculate the hashvalue
+4. Append the hashvalue to the actual data
+5. Store the results of step 4.
+
+
+```obj-c
+	// Allocate a buffer to hold the digest, and perform the digest.
+	NSMutableData* actualData = [getData]; //get the data
+ 	//get the key from the keychain
+	NSData* key = [getKey];	
+    NSMutableData* digestBuffer = [NSMutableData dataWithLength:CC_SHA256_DIGEST_LENGTH];
+    CCHmac(kCCHmacAlgSHA256, [actualData bytes], (CC_LONG)[key length], [actualData
+     bytes], (CC_LONG)[actualData length], [digestBuffer mutableBytes]);
+     [actualData appendData: digestBuffer];
+```
 
+When verifying the HMAC with CC:
+1. Extract the message and the hmacbytes as separate `NSData` .
+2. Repeat step 1-3 of generating an hmac on the `NSData`.
```

Comment:
Shouldn't this be the NSMutableData instead of NSData according to the example below?